mesosclient/mesos.go

```diff
@@ -0,0 +1,41 @@
+package mesosclient
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mesos/mesos-go/api/v1/lib/httpcli"
+	"github.com/mesos/mesos-go/api/v1/lib/httpcli/httpmaster"
+	"github.com/mesos/mesos-go/api/v1/lib/master"
+	"github.com/mesos/mesos-go/api/v1/lib/master/calls"
+)
+
+// Client is a Mesos API client.
+type Client struct {
+	sender calls.Sender
+}
+
+// NewClient builds a new Client object that queries a Mesos API endpoint at
+// the given base URL. The baseURL parameter should have the form
+// "http://host:port".
+func NewClient(baseURL string) *Client {
+	endpoint := httpcli.Endpoint(fmt.Sprintf("%s/api/v1", baseURL))
```

Comment:
I'm wondering about other config options, like multiple URLs, timeouts, TLS, etc. But I guess that is something for another PR/when we care about it.

Reply:
`mesos-go` doesn't really support that stuff (although we could probably use `context.Context` for timeouts) so we'd have to add it ourselves. I'd prefer to wait until we have a better idea of how and where we'll be calling Mesos before doing that.